tsc/frr/sonic_frr_update_process.md

```diff
@@ -0,0 +1,79 @@
+-   Rotate SONiC FRR maintenance duty among repo maintainer org and others (BRCM, MSFT, Alibaba, NVDA, DELL)
+-   Responsibility of SONiC FRR release maintainer
+	-   Default 12 months assignment
+	-   Upgrade FRR version in Nov release, resolve SONiC FRR upgrade integration issues
+	-   Triage and fix SONiC FRR issues when applicable. Fix may come from SONiC contributors or from FRR community, maintainer is responsible to drive the fix to unblock SONiC community
+	-   Submit fixes to FRR project, submit new FRR topo test to FRR project if there is a gap
+	-   Release maintainer to subscribe to FRR project, and be the FRR Point-of-Contact on behalf of SONiC
+	-   Bring in FRR vulnerabilities and critical patches to SONiC
+  -   If there is a need for May release upgrade due to feature requirement or other reasons, it should be requested in the May release plan and get a consensus with the FRR release maintainer. The person or organization requested the upgrade will be responsible of carrying out FRR upgrade with the same FRR upgrade process described in this proposal. FRR release maintainer will need to work closely with the person or organization to oversee the mid term upgrade, he/she will continue to assume the role of FRR release maintainer until the end of his/her term
+
+# SONiC FRR vulnerability and patch upgrade in between SONiC releases
+
+-   FRR CVE Fixes
+	-   Reference nvd.nist.gov cvss v3.x [rating](https://nvd.nist.gov/vuln-metrics/cvss#)
+	-   Bring in Critical and High FRR CVE patches into SONiC
+	-   Sample of [Critical](https://nvd.nist.gov/vuln/search/results?form_type=Advanced&results_type=overview&search_type=all&isCpeNameSearch=false&cpe_vendor=cpe%3A%2F%3Afrrouting&cpe_product=cpe%3A%2F%3A%3Afrrouting&cvss_version=3&cvss_v3_severity=CRITICAL), sample of [High](https://nvd.nist.gov/vuln/search/results?form_type=Advanced&results_type=overview&search_type=all&isCpeNameSearch=false&cpe_vendor=cpe%3A%2F%3Afrrouting&cpe_product=cpe%3A%2F%3A%3Afrrouting&cvss_version=3&cvss_v3_severity=HIGH)
+	-   There are regular security scan in Azure pipeline, CVEs will be filed to sonic-buildimage/issues
+	-   SONiC FRR release maintainer should subscribe to nvd.nist.gov for FRR alerts
+	-   Need a process to bring in CVEs to earlier SONiC releases too (open to suggestions)
+
+-   Patch FRR Bug Fixes
+	-   SONiC FRR release maintainer should subscribe to FRR project to bring in critical patch which is applicable to SONiC
+
+# SONiC FRR Upgrade Steps
+-   Create sonic-frr branch for the target FRR version
+	-   Contact release manager
+-   Find new package dependencies
+	-   Upload newly required packages to a common location (Azure)
+-   Submodule update to new FRR commit id
+-   Code changes
+    -   Version change in Makefiles
+    -   New Makefiles for new packages (if any)
+    -   Port patches
+	-   Evaluate whether existing FRR patches still applicable to new FRR version
```

Comment:
Is this not a FRR community activity? I believe, we need to worry about SONiC specific changes on top of the FRR community code. Hope we clearly differentiate the SONiC specific patches vs FRR patches in the file-name or by some other means.

Reply:
I'm trying to understand the issues that we'll encounter if we discard all local FRR code and it's associated patches when we migrate from one version to another FRR release and apply only the SONiC specific changes on top of the FRR code. Am I missing anything basic here?

Reply:
@venkatmahalingam , the sign-off details in the commit is supposed to have the original credentials (as mentioned in the next line) and should be sufficient to identify whether the changes were committed to sonic-frr or frr. Ideally, frr patches shouldn't require porting when moving to next frr version.
> Apply the old patches into new FRR version, and generate new patch files. Keep original credentials